### PR TITLE
[UPDATE] notification in slack on user create

### LINKED ISSUE #736: Send slack notification when new user is created
Currently, a Slack notification is sent to a Slack channel when a survey is completed.
The goal of this issue is to also send a notification when a user is created.

The message should be like:
```
New user with role XXX created at {date}. 
There are X admins, Y teachers and Z students
Total users: X + Y + Z
This code should be dynamic

### PR DESCRIPTION
**Pull request by nikhilbhatt**
fix #736 

# Notifications will be send on slack on new user register.

## Type of changes

Please delete options that are not relevant.

- [ ] **Bug fix**
- [x] **New feature**
- [ ] **Breaking change**

---

## Checklist:

- [x] My code is understandable
- [x] My code follows the styleguide of this project
- [x] My changes don't generate new warnings/errors
- [x] I have performed a self-review of my own code (build/lint/tests)

  ### Backend

  Please delete options that are not relevant.

  - [x] **Model testing.**
  - [ ] **Request testing and swagger doc update.**
  - [ ] **System testing.**
  - [ ] **No tests required.**

  ### Frontend

  - Components:

    - [ ] I have implemented/updated unit tests
    - [ ] I have implemented/updated Storybook documentation

  - Pages:
    - [ ] I have implemented/updated E2E tests

### CODE REVIEW

**Review by danielnottingham — APPROVED** [review]

**Review by philipeleandro — APPROVED** [review]
@nikhilbhatt You can try to specify a little bit better on PR description what you did, how we can test what you did. Just a few tips. Good job!

File: backend/app/models/user.rb [hunk 1]
```diff
@@ -36,4 +36,27 @@ class User < ApplicationRecord
   delegate :admin?, to: :role
   delegate :teacher?, to: :role
   delegate :student?, to: :role
+
+  after_create :slack_notify
+
+  private
+
+  def slack_notify
```
Comment (edimossilva) [inline backend/app/models/user.rb:44]: If we were going with this approach I would suggest a more descriptive name like `slack_notify_after_creation`

File: backend/app/models/user.rb [hunk 2]
```diff
@@ -36,4 +36,27 @@ class User < ApplicationRecord
...
+    SlackService.call(message)
+  end
+
+  def message
```
Comment (edimossilva) [inline backend/app/models/user.rb:48]: `message` is also a non specific name for a method like this, I rather go with something like `after_creation_message`

File: backend/app/models/user.rb [hunk 3]
```diff
@@ -36,4 +36,27 @@ class User < ApplicationRecord
...
+    I18n.t('users.create', **translate_arguments)
+  end
+
+  def translate_arguments
+    {
+      role: role.role_type.humanize,
+      date: created_at,
+      admin: Role.find_by(role_type: Role::ADMIN)&.users_count || 0,
```
Comment (edimossilva) [inline backend/app/models/user.rb:56]: In this case I would refactor this to scopes in the Role class. This scope would receive a role as parameter

**Review by edimossilva — CHANGES_REQUESTED** [review]
Hey @nikhilbhatt, I was thinking in some approach like what is done [here](https://github.com/espoo-dev/espoo-dev/blob/main/backend/app/controllers/api/v1/answers_controller.rb#L3), the inspiration come from [this](https://www.youtube.com/watch?v=XH1fbvexqyU)

We are trying to keep the the model a bit more isolated from the business logic. One of the ways of doing this is to have business logic in the services layer. 

Sorry for not putting this in the issue description, this approach is brand new to the project.

I am also letting some comments with my some input I consider useful to provide some feedback

**Review by edimossilva — APPROVED** [review]
Hey great work, thanks!

**Pull request merged**